From 25229773a9d4472235278bb45f75439e56630cee Mon Sep 17 00:00:00 2001
From: Joe Slater <joe.slater@windriver.com>
Date: Wed, 22 Jul 2020 13:31:11 -0700
Subject: [PATCH] libdnf: allow reproducible binary builds

Use a dummy directory for test data if not built WITH_TESTS.  Allow for overriding
TESTDATADIR, since the default is guaranteed to be wrong for target builds.

Upstream-Status: Pending

Signed-off-by: Joe Slater <joe.slater@windriver.com>
---
 CMakeLists.txt | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 548a9137..c378e7d9 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -132,7 +132,12 @@ add_definitions(-DGETTEXT_DOMAIN=\\"libdnf\\")
 add_definitions(-DG_LOG_DOMAIN=\\"libdnf\\")
 
 # tests
-add_definitions(-DTESTDATADIR=\\"${CMAKE_SOURCE_DIR}/data/tests\\")
+if(NOT WITH_TESTS)
+  set(TEST_DATA_DIR "/notests")
+elseif(NOT DEFINED TEST_DATA_DIR)
+  set(TEST_DATA_DIR "${CMAKE_SOURCE_DIR}/data/tests")
+endif()
+add_definitions(-DTESTDATADIR=\\"${TEST_DATA_DIR}\\")
 
 # librhsm
 if(ENABLE_RHSM_SUPPORT)
